### DISCUSSION TITLE
USB charging of uv5R plus devices

### DISCUSSION THREAD

**Discussion by mentanah** [egzumer/uv-k5-firmware-custom#371] [Show and tell]
Dear all, I just stumbled over a smart solution for charging radios without USB charge capability like the UV5R plus radio and want to share it with you:

[Link](https://de.aliexpress.com/item/1005006065599445.html?spm=a2g0n.detail.1000014.14.47e13fc3LDZuqy&gps-id=platformRecommendH5&scm=1007.18499.332340.0&scm_id=1007.18499.332340.0&scm-url=1007.18499.332340.0&pvid=0ec08549-2bbd-4ecc-8601-bffd160237a3&_t=gps-id:platformRecommendH5,scm-url:1007.18499.332340.0,pvid:0ec08549-2bbd-4ecc-8601-bffd160237a3,tpp_buckets:668%232846%238114%231999&pdp_npi=4%40dis%21EUR%2111.41%215.47%21%21%2112.30%21%21%40211b600417042079229165894e1e0e%2112000035570533614%21rec%21DE%21905132458%21)

![Screenshot_2024-01-02-16-09-38-592_nextapp fx-edit](https://github.com/egzumer/uv-k5-firmware-custom/assets/153547177/03ca1868-8e68-4493-864f-6479e41ef934)

**Comment by OE1MWW** [comment]
unfortunately the 2200 mAh battery is too thick, does not fit, clips can not open that wide

**Reply by OE1MWW** [reply]
btw. I don't own one, but I know someone, who was eager to have it... now well stored in the drawer :-)

**Comment by egzumer** [comment]
You still have to carry the damned charger. Why not just buy K6? It is literally the same radio.

**Reply by mentanah** [reply]
Just a hint, the orange knop is for me a nogo.
 When traveling you have to carry an USB charger and cable as well, right?

**Reply by egzumer** [reply]
Buy a second K6 then, take a battery and the knob from 5R.

> When traveling you have to carry...

Yes, but just one, the one that you can also charge your phone with, and many other devices.
...and...🥁🥁🥁🥁🥁... you can still borrow a charger from anybody else if you forgot yours.

**Reply by JuantAldea** [reply]
There's a battery replacement with USB-charging. Check it out on AliExpress.

**Reply by egzumer** [reply]
Yeah, and it is more expensive than a K6.

**Reply by JuantAldea** [reply]
It is indeed :laughing:

There's also a printable 2s18650 case, maybe one could fit a little BMS there.

**Reply by pietss** [reply]
The printing solution (thingiverse) takes time and patience but can be rewarding.
However, if there is no rush, we can assume that in some time we will be able to find these casings for AA batteries, this one is for the UV-5R https://ae01.alicdn.com/kf/Sdc32af323f504af382a32c1e61083438z.jpg (Aliexpress)

**Reply by ErikS-web** [reply]
I already have a 3D printed case. And now I'm looking for the parts to make the electrical contacts.

**Reply by pietss** [reply]
You could use these (random sellers links...) 
I used the brass plate (not the one in the link but similar) for the positive and negative in contact with the radio body (photo in thingiverse - mine is the one I painted) 

https://it.aliexpress.com/item/1005005972222025.html?spm=a2g0o.productlist.main.7.5da649331HbiTv&algo_pvid=64af96b1-b93c-4d81-acc5-a4d4d41dcd4d&algo_exp_id=64af96b1-b93c-4d81-acc5-a4d4d41dcd4d-3&pdp_npi=4%40dis%21EUR%213.25%210.46%21%21%2124.79%21%21%40211b600917043509657241753e196d%2112000035151576176%21sea%21IT%210%21AB&curPageLogUid=Um8IgzIUKWJP

https://it.aliexpress.com/item/1005004697690882.html?spm=a2g0o.productlist.main.37.4a104564xSLSap&algo_pvid=c271c57a-d8c1-4707-8087-12bdd7a3b9b8&algo_exp_id=c271c57a-d8c1-4707-8087-12bdd7a3b9b8-18&pdp_npi=4%40dis%21EUR%2130.17%213.02%21%21%21230.31%21%21%40211b5db316992620404238819eac27%2112000030137734875%21sea%21IT%210%21AB&curPageLogUid=dNQsrbqtzgoS

For the connection and release system from the radio body you can use a negative pole plate since it is equipped with a spring, to be honest here it seems you need to do a bit of testing before definitively gluing, I think for a couple of reasons, probably the stl file in those details is not precise enough, and also because we are talking about tenths of a millimeter and from my modest experience with a 3D printer it is difficult to obtain results of sufficient precision when just a tenth of error can be problematic, and perhaps also because the spring I used is placed on a plate which therefore limits the excursion a little, however these are tests that must be done empirically, for example in my case I had to use a cutter to further shape the two hooks.

**Comment by ErikS-web** [comment]
Or buy a 

"QUANSHENG UV-K5 UV-K6 Walkie Talkie Battery Thicken Real 2600mAh/3500mAh Type-C Charge Battery For UVK5 Radio Parts 18650 Cell"

![USB-C battery](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/68474238-bffb-4148-864d-107eedae495d)

Prices 15.. 20 euro.
I think this will fit on UV5R-plus as well.

**Reply by ErikS-web** [reply]
Tested... **a UVK5 battery fits directly on a UV5RPlus.** 
So buying this 2600mA is not a bad idea:
- more capacity. 
- can be charged on original table charger
- can be charged with USB-C cable.

The other way around is only possible if a tab in the middle is mechanically removed on the originally-battery (2200mA) from the UV5RPlus.
